src/docfx/build/moniker/MonikerProvider.cs

```diff
@@ -43,6 +43,8 @@ public MonikerProvider(Config config, BuildScope buildScope, MetadataProvider me
             _monikerOrder = GetMonikerOrder(monikerDefinition);
         }
 
+        public (List<Error>, MonikerList) Validate(SourceInfo<string?>[] monikers) => _rangeParser.Validate(monikers);
```

Comment:
make `MonikerProvider.GetFileLevelMonikers` return the right moniker list for redirection, so code that depend on that method works. (Like publish url conflict validation, consider a test to cover multiple versions of the same file with different redirect URLs)

Reply:
Actually, I do not think the current model could fit this requirement.
`MonikerProvider.GetFileLevelMonikers(FilePath)` and `FilePath` of the redirection will be the same for different monikers, not like the physical files, for them different files would always end up with different monikers